muxrpc/handlers/tunnel/server/members.go

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: 2021 The NGI Pointer Secure-Scuttlebutt Team of 2020/2021
+//
+// SPDX-License-Identifier: MIT
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/ssb-ngi-pointer/go-ssb-room/v2/internal/network"
+	"github.com/ssb-ngi-pointer/go-ssb-room/v2/roomdb"
+	"go.cryptoscope.co/muxrpc/v2"
+	refs "go.mindeco.de/ssb-refs"
+)
+
+type Member struct {
+	ID refs.FeedRef `json:"id"`
+}
+
+func (h *Handler) members(ctx context.Context, req *muxrpc.Request, snk *muxrpc.ByteSink) error {
+	peer, err := network.GetFeedRefFromAddr(req.RemoteAddr())
+	if err != nil {
+		return err
+	}
+
+	pm, err := h.config.GetPrivacyMode(ctx)
+	if err != nil {
+		return fmt.Errorf("running with unknown privacy mode: %w", err)
+	}
+
+	if pm == roomdb.ModeCommunity || pm == roomdb.ModeRestricted {
+		_, err := h.membersdb.GetByFeed(ctx, *peer)
+		if err != nil {
+			return fmt.Errorf("external user are not allowed to list members: %w", err)
```

Comment:
Is there a test for this also? Seems important.

Reply:
There isn't, in general we don't seem to have tests for security checks like this one in other similar endpoints. I will try to add one.

Reply:
Is this in any way readable? I am under the impression this tests became very messy. [d3a77be](https://github.com/ssbc/go-ssb-room/pull/340/commits/d3a77be89452ee66a831b9efc8881599e4181a2f)